### PR TITLE
add first draft or daily report

### PR DESCRIPTION
**Pull request by ljwoodley**
Do not merge. RMD report in progress.

### CODE REVIEW

File: daily_contact_tracing_report.R [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  count(person_type, name = 'quarantined') %>% 
+  arrange(desc(quarantined))
+
+isolated_and_quarantined <- isolated_cases %>% 
+  full_join(quarantined_contacts, by = "person_type") %>%  
+  mutate_at(vars(isolated, quarantined),  ~replace(., is.na(.), 0)) %>% 
+  mutate(total = isolated + quarantined) %>% 
+  arrange(desc(total))
+
+write_csv(count_of_cases_by_person_type, "count_of_cases_by_person_type.csv")
```
Comment (pbchase) [inline daily_contact_tracing_report.R:82]:
```suggestion
write_csv(count_of_cases_by_person_type, "count_of_cases_for_all_time_by_person_type.csv")
```

File: daily_contact_tracing_report.R [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    person_type == 'Jax' ~ 'UF Employee or Student Located in Jacksonville',
+    is.na(contact_ufid) ~ "Non-UF",
+    !is.na(contact_ufid) & is.na(person_type) ~ "Unknown")) %>% 
+  count(person_type, name = 'quarantined') %>% 
+  arrange(desc(quarantined))
+
+isolated_and_quarantined <- isolated_cases %>% 
+  full_join(quarantined_contacts, by = "person_type") %>%  
+  mutate_at(vars(isolated, quarantined),  ~replace(., is.na(.), 0)) %>% 
+  mutate(total = isolated + quarantined) %>%
```
Comment (pbchase) [inline daily_contact_tracing_report.R:79]:
```suggestion
  mutate(n = isolated + quarantined) %>% 
  select(person_type, n) %>%
```

File: daily_contact_tracing_report.R [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+         contact_ll_firstname, contact_ll_lastname, contact_ufid, contact_released_from_quarantine,
+         contact_released_from_quarantine_date)
+
+# Are there empty case_report_datetime? 
+empty_case_report_datetime <- cases_project_read %>% 
+  filter(redcap_event_name == "case_investigation_arm_1") %>%   
+  filter(is.na(redcap_repeat_instrument)) %>%  
+  filter(is.na(case_report_datetime))
+
+count_of_cases_by_person_type <- case_records %>%
```
Comment (pbchase) [inline daily_contact_tracing_report.R:51]:
```suggestion
count_of_cases_for_all_time_by_person_type <- case_records %>%
```

File: daily_contact_tracing_report.R [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  select(record_id, case_emp_ufid, case_name_first, case_name_last, case_dob,
+         person_type) %>%  
+  count(person_type) %>%
```
Comment (pbchase) [inline daily_contact_tracing_report.R:54]:
```suggestion
  count(person_type) %>% 
  select(person_type, n) %>%
```

**Comment by pbchase** [comment]
I made a few suggested changes to the code that reflect things said in the meeting and shortly thereafter. 

Could you also add date_stamps to the output file names? e.g.

```
filename = paste0('basename_', format(script_run_time, '%Y%m%d%H%M%S'), '.csv')
```

**Comment by pbchase** [comment]
Please also know that we will very quickly need to go to full automation.

To that end, please make these additions:

1. Add the write of the output file.  I think this is a good first order answer:

```
filename = paste0('count_of_cases_for_all_time_by_person_type_', format(script_run_time, '%Y%m%d%H%M%S'), '.csv')
write_csv(count_of_cases_for_all_time_by_person_type, filename)
put_result <- system(paste('bash put_file.sh', filename, 'data_out/redcap/'), intern = TRUE)
unlink(filename)

filename = paste0('isolated_and_quarantined_', format(script_run_time, '%Y%m%d%H%M%S'), '.csv')
write_csv(isolated_and_quarantined, filename)
put_result <- system(paste('bash put_file.sh', filename, 'data_out/redcap/'), intern = TRUE)
unlink(filename)
```

2. Add an email targeted for you and me summary of what the script did.

3. Add a cron entry for this new script.

**Comment by ljwoodley** [comment]
Is an RMD report still needed or stick with only the R script?

**Comment by pbchase** [comment]
> Is an RMD report still needed or stick with only the R script?

Stick with R script for now. 

Consider also how you want to deliver the epicurve data. We want to be consistent with what we show in the weekly and the daily, so I am thinking we need a common code base for the epicurve dataset. For the weekly report, we will need to lock the data to last Sunday at 23:59:59. For the daily we need to show a different time frame. Jerne has yet to say how fresh the daily report data should be, but I am thinking up through 23:59:59 yesterday.

**Pull request merged**